contracts/integrations/custom/CustomIntegrationLiquity.sol

```diff
@@ -0,0 +1,232 @@
+
+    /**
+     * The list of addresses of the IERC-20 tokens mined as rewards during the strategy
+     *
+     * hparam  _data                      Address provided as param
+     * @return address[] memory           List of reward token addresses
+     */
+    function _getRewardTokens(
+        address /* _data */
+    ) internal pure override returns (address[] memory) {
+        // no LQTY rewards for lending or borrowing
+        return new address[](0);
+    }
+
+    /* ============ External Functions ============ */
+
+    /**
+     * The tokens to be purchased by the strategy on enter according to the weights.
+     * Weights must add up to 1e18 (100%)
+     *
+     * hparam  _data                      Address provided as param
+     * @return _inputTokens               List of input tokens to buy
+     * @return _inputWeights              List of weights for the tokens to buy
+     */
+    function getInputTokensAndWeights(
+        bytes calldata /* _data */
+    ) external pure override returns (address[] memory _inputTokens, uint256[] memory _inputWeights) {
+        address[] memory inputTokens = new address[](1);
+        // inputTokens[0] is address(0) (ether)
+        uint256[] memory inputWeights = new uint256[](1);
+        inputWeights[0] = 1e18; // 100%
+        return (inputTokens, inputWeights);
+    }
+
+    /**
+     * The tokens to be received on exit.
+     *
+     * hparam  _strategy                  Strategy address
+     * hparam  _data                      Bytes data
+     * hparam  _liquidity                 Number with the amount of result tokens to exit
+     * @return exitTokens                 List of output tokens to receive on exit
+     * @return _minAmountsOut             List of min amounts for the output tokens to receive
+     */
+    function getOutputTokensAndMinAmountOut(
+        address _strategy,
+        bytes calldata, /* _data */
+        uint256 /* _liquidity */
+    ) external view override returns (address[] memory exitTokens, uint256[] memory _minAmountsOut) {
+        address[] memory outputTokens = new address[](1);
+        // outputTokens[0] is address(0) (ether)
+        uint256[] memory outputAmounts = new uint256[](1);
+        outputAmounts[0] = _getCollateral(_strategy); // TODO: deal with fees
+        return (outputTokens, outputAmounts);
+    }
+
+    /**
+     * The price of the result token based on the asset received on enter
+     *
+     * hparam  _data                      Bytes data
+     * hparam  _tokenDenominator          Token we receive the capital in
+     * @return uint256                    Amount of result tokens to receive
+     */
+    function getPriceResultToken(
+        bytes calldata _data,
+        address /* _tokenDenominator */
+    ) external pure override returns (uint256) {
+        (, uint amountLUSDToBorrow) = abi.decode(_data, (address, uint));
+        return amountLUSDToBorrow; // we receive the exact number of LUSD tokens that we requested
```

Comment:
See note above about using LUSD as the result token. Is this ok?

Reply:
we need to do something special here. Basically we want to return in total the value of the collateral minus the value of the LUSD taken as debt.

So the price per token should be : (Collateral in LUSD - Debt in LUSD)/ debt in LUSD

Reply:
I don't believe this function knows how much `collateralInLUSD` there is, however?

Reply:
you would need the strategy address to fetch it right?